docs(widget-library): EA docs for widget library (#4595)
getsentry/sentry-docs: src/docs/product/dashboards/widget-library/index.mdx

- Search condition 2: `!event.type:error release.version:{another_version}`

You can also compare performance before/and after a certain version with the following conditions:

- Search condition 1: `!event.type:error release.version:<{version}`
- Search condition 2: `!event.type:error release.version:>={version}`

### Response thresholds

The "Distribution Duration" widget shows the spread of transaction duration times. This visualization can be modified to show if transaction durations are within certain target thresholds and present them as "satisfactory", "tolerable", or "frustrating" transactions. To create this type of visualization, change the "Visualization Display" to "Area Chart". Area charts stack results and are more appropriate for results that are cumulative.

- Visualization Display: `Area Chart`

Set one of the y-axis values to `count()` and remove the other two axes.

- Y-Axis: `count()`

To the first query, add the search condition for satisfactory transaction duration (this example uses 300ms as the satisfactory response threshold).

- Search condition 1: `!event.type:error transaction.duration:<300`

> Add another query for the tolerable condition by clicking the "Add Query" button. In this example, we're using the Apdex definition where tolerable response times are between the satisfactory threshold and four times the satisfactory threshold.

Reviewer: ```suggestion
Add another query for the tolerable condition by clicking the "Add Query" button. In this example, we're using the [Apdex](/product/performance/metrics/#apdex) definition where tolerable response times are between the satisfactory threshold and four times the satisfactory threshold.
```

I think we should link users to our own Apdex defintion for more info

- Search condition 2: `!event.type:error (transaction.duration:<1200 AND transaction.duration:>=300)`

Finally, add a third query for for the frustrating transaction duration:

- Search condition 3: `!event.type:error transaction.duration:>=1200`

The chart now shows cumulative counts at different response time thresholds.

![Response Threshold Widget.](dashboard-response-threshold-widget.png)

### Top issues

You can track the impact of your issues by counting the error events associated with each unique issue. Modify the "Latest Unresolved Issues" widget by selecting the "All Events" data set. Update the search condition to filter for error events:

- Data Set: `All Events`
- Search condition: `event.type:error`

Update columns to group by `issues` and apply `count()` as the aggregating function. Order count in descending order so the widget displays issues with the most error events:

- Columns: `issue, count(), any(message)`
- Sort by: `count() desc`

### Unique issues by projects

> Monitoring the cardinality of issues alongside the total number of errors in your projects can give you a sense of the impact of issues in your project. For example, low cardinality in issues and a high number of total errors indicates you have a few very high impact issues. Modify the "Latest Unresolved Issues" widget by selecting the "All Events" data set. Update the search condition to filter to error events:

Reviewer 2: I think just defining cardinality here so we can keep on using it makes the most sense.
```suggestion
Monitoring the cardinality of issues (the number of unique errors in an issue) alongside the total number of errors in your projects can give you a sense of the impact of issues in your project. For example, low cardinality  in issues and a high number of total errors indicates you have a few very high impact issues. Modify the "Latest Unresolved Issues" widget by selecting the "All Events" data set. Update the search condition to filter to error events:
```

Author: Updated here https://github.com/getsentry/sentry-docs/pull/4595/commits/14fb76a203a5e4c49c0be6affd312009de562d9f
(definition slightly different)

---
title: Widget Library

> sidebar_order: 1

Reviewer 2: ```suggestion
sidebar_order: 20
```

> redirect_from:
>  - /product/dashboards/customize-dashboards/

Reviewer 2: This redirect is only needed for the Custom Dashboards page
```suggestion
```

> description: "Learn how to edit or create additional dashboards to suit your organization's needs."

Reviewer 2: ```suggestion
description: "Learn about Sentry's library of prebuilt widgets that you can use in your custom dashboards."
```

---

<Note>

This feature is available only if your organization is on either a Business or Trial plan.

</Note>

> The Widget Library contains a collection of prebuilt widgets you can add to your [Custom Dashboards](/product/dashboards/customize-dashboards/). To access the Widget Library, click on "Add Widget(s)" and switch to the "Widget Library" tab on the modal that appears.

Reviewer 2: To avoid having to capitalize this every time we talk about it forever, I'm going to lower case this unless we're specifically calling it out in the UI.

```suggestion
The widget library contains a collection of prebuilt widgets you can add to your [custom dashboards](/product/dashboards/customize-dashboards/). To access it, click on "Add Widget(s)" and switch to the "Widget Library" tab in the modal that appears.
```

> The following widgets can be found in the library:

Reviewer 2: Switching this to active voice

```suggestion
The library includes the following widgets:
```

> - **Distribution Duration**: A multi-y axis chart visualizing the distribution of transaction duration across various percentiles (p50, p75 and p95)

Reviewer 2: ```suggestion
- **Distribution Duration**: A multi-y axis chart visualizing the distribution of transaction duration across various percentiles (p50, p75, and p95)
```

> - **High Throughput Transactions**: The top 5 transactions with the largest number of events

Reviewer 2: ```suggestion
- **High Throughput Transactions**: The top five transactions with the largest number of events
```

- **LCP by Country**: World map showing the p75 of page load times for each country
- **Miserable Users**: The total number of unique users who have experienced slow transactions (transaction duration greater than 1200ms)
- **Slow vs Fast Transactions**: Bar chart comparing the percentage of transactions that are over 300ms (slow) and under 300ms (fast)
- **Latest Unresolved Issues**: A table of unresolved issues ordered by the most recently seen issues

> - **Top Unhandled Error Types**: The top 5 most frequently encountered unhandled errors

Reviewer 2: ```suggestion
- **Top Unhandled Error Types**: The top five most frequently encountered unhandled errors
```

- **Users Affected by Errors**: A comparison of the total number of errors and the number of unique users affected by the errors

> You can change the title, queries, fields, visualization types and sort order of these prebuilt widgets to suit your use case by clicking on the context menu on the widget and selecting "Edit Widget".

Reviewer 2: ```suggestion
You can change the title, queries, fields, visualization types, and sort order of these prebuilt widgets to suit your use case by clicking the context menu on the widget and selecting "Edit Widget".
```

## Example Use Cases

### Performance across release versions

> To monitor how your app is performing on a certain release, you can modify the **Distribution Duration** widget by adding the release version to the search condition.

Reviewer 2: Using bold on these widget names makes sense in the context of the bullet list to make them pop, but here, we're just treating them as UI elements so they're no-bold w quotation marks.

```suggestion
To monitor how your app is performing on a certain release, you can modify the "Distribution Duration" widget by adding the release version to the search condition.
```

- Search condition 1: `!event.type:error release.version:{version}`

To compare performance with another release version, you can add another query by clicking the "Add Query" button and querying for a different release version.

- Search condition 2: `!event.type:error release.version:{another_version}`

You can also compare performance before/and after a certain version with the following conditions:

- Search condition 1: `!event.type:error release.version:<{version}`
- Search condition 2: `!event.type:error release.version:>={version}`

### Response thresholds

> The **Distribution Duration** widget shows how the spread of transaction duration times. This visualization can be modified to show if transaction durations are within certain target thresholds and present them as "satisfactory", "tolerable" or "frustrating" transactions. Change the "Visualization Display" to "Area Chart". Area charts stack results and are more appropriate for results that are cummulative.

Reviewer 2: ```suggestion
The "Distribution Duration" widget shows the spread of transaction duration times. This visualization can be modified to show if transaction durations are within certain target thresholds and present them as "satisfactory", "tolerable", or "frustrating" transactions. To create this type of visualization, change the "Visualization Display" to "Area Chart". Area charts stack results and are more appropriate for results that are cumulative.
```

- Visualization Display: `Area Chart`

Set one of the y-axis values to `count()` and remove the other two axes.

- Y-Axis: `count()`

To the first query, add the search condition for satisfactory transaction duration (this example uses 300ms as the satisfactory response threshold).

> - Search Condition 1: `!event.type:error transaction.duration:<300`

Reviewer 2: Just keeping this consistent with the first example since this doesn't represent any UI text. 
```suggestion
- Search condition 1: `!event.type:error transaction.duration:<300`
```

> Add another query by clicking the "Add Query" button for the tolerable condition (using the Apdex definition where tolerable response times are between the satisfactory threshold and 4 times the satisfactory threshold).

Reviewer 2: ```suggestion
Add another query for the tolerable condition by clicking the "Add Query" button. In this example, we're using the Apdex definition where tolerable response times are between the satisfactory threshold and four times the satisfactory threshold.
```

> - Search Condition 2: `!event.type:error (transaction.duration:<1200 AND transaction.duration:>=300)`

Reviewer 2: ```suggestion
- Search condition 2: `!event.type:error (transaction.duration:<1200 AND transaction.duration:>=300)`
```

Finally, add a third query for for the frustrating transaction duration.

> - Search Condition 3: `!event.type:error transaction.duration:>=1200`

Reviewer 2: ```suggestion
- Search condition 3: `!event.type:error transaction.duration:>=1200`
```

> The graph now shows cumulative counts at different respone time thresholds.

Reviewer 2: ```suggestion
The chart now shows cumulative counts at different response time thresholds.
```

![Response Threshold Widget.](dashboard-response-threshold-widget.png)

### Top issues

> You can track the impact of your issues by counting the error events associated with each unique issue. Modify the **Latest Unresolved Issues** widget by selecting the "All Events" dataset. Update the search condition to filter to error events.

Reviewer 2: ```suggestion
You can track the impact of your issues by counting the error events associated with each unique issue. Modify the "Latest Unresolved Issues" widget by selecting the "All Events" data set. Update the search condition to filter for error events.
```

- Data Set: `All Events`

> - Search Condition: `event.type:error`

Reviewer 2: ```suggestion
- Search condition: `event.type:error`
```

> Update columns to group by `issues` and apply `count()` as the aggregating function. Order count in descending order to get issues with the most error events.

Reviewer 2: ```suggestion
Update columns to group by `issues` and apply `count()` as the aggregating function. Order count in descending order so the widget displays issues with the most error events.
```

- Columns: `issue, count(), any(message)`
- Sort by: `count() desc`

### Unique issues by projects

> Monitoring the cardinality of issues across alongside the total number of errors your projects can give you a sense of the impact of issues in your project. For example, low cardinality in issues and a high number of total errors indicates you have a few very high impact issues. Modify the **Latest Unresolved Issues** widget by selecting the "All Events" dataset. Update the search condition to filter to error events.

Reviewer 2: Is "cardinality" a common concept in programming that we can expect everyone to understand?

```suggestion
Monitoring the cardinality of issues alongside the total number of errors in your projects can give you a sense of the impact of issues in your project. For example, low cardinality in issues and a high number of total errors indicates you have a few very high impact issues. Modify the "Latest Unresolved Issues" widget by selecting the "All Events" data set. Update the search condition to filter to error events.
```

- Data Set: `All Events`

> - Search Condition: `event.type:error`

Reviewer 2: ```suggestion
- Search condition: `event.type:error`
```

> Update columns to group by `project` and apply `count()` and `count_unique(issue)` as the aggregating functions. Order count in descending order to get projects with the most error events.

Reviewer 2: ```suggestion
Update columns to group by `project` and apply `count()` and `count_unique(issue)` as the aggregating functions. Order count in descending order so the widget displays projects with the most error events.
```
